### PR TITLE
[Merged by Bors] - chore: Move sign of power lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
* Move the sign of power lemmas from `Algebra.Parity` to `Algebra.GroupPower.Order`.
* For this to work, I must swap the order of import between `Algebra.GroupPower.Order` and `Algebra.Parity`. This means that I need to weaken one `assert_not_exists` to allow importing `Data.Set.Defs`. This is inconsequential.
* Use them to golf and deprecate the `bit0`/`bit1` lemmas
* Delete the deprecated `pow_bit0_abs`, `pow_bit0_pos_of_neg`, `pow_bit1_neg`

---

- [x] depends on: #12260
- [x] depends on: #12288
- [x] depends on: #12289
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Parity.lean [hunk 1]
```diff
@@ -273,24 +276,23 @@ alias ⟨Even.exists_bit0, _⟩ := even_iff_exists_bit0
 
 section Semiring
 
-variable [Semiring α] [Semiring β] {m n : α}
+variable [Semiring α] [Semiring β] {a b : α} {m n : ℕ}
```
Comment (eric-wieser) [inline Mathlib/Algebra/Parity.lean:279]: Can you do this rename in a standalone PR with F2?

Comment (YaelDillies) [reply Mathlib/Algebra/Parity.lean:279]: #12260 but I think this is split-abuse. This PR is already split off #11855 and #11863

**Review by Parcly-Taxel — APPROVED** [review]

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]
I think this is doing a bit too much at once, but I'll try to split it up a bit myself. Please give me a day or two.

**Comment by YaelDillies** [comment]
I did the minimal amount of fixes, but if you can find something smaller I'm all ears!

File: Mathlib/Algebra/GroupPower/Order.lean [hunk 2]
```diff
@@ -298,22 +297,9 @@ theorem sq_pos_of_pos (ha : 0 < a) : 0 < a ^ 2 := pow_pos ha _
...
 
-theorem pow_bit0_pos_of_neg (ha : a < 0) (n : ℕ) : 0 < a ^ bit0 n := by
-  rw [pow_bit0']
-  exact pow_pos (mul_pos_of_neg_of_neg ha ha) _
-#align pow_bit0_pos_of_neg pow_bit0_pos_of_neg
-
-theorem pow_bit1_neg (ha : a < 0) (n : ℕ) : a ^ bit1 n < 0 := by
-  rw [bit1, pow_succ']
-  exact mul_neg_of_neg_of_pos ha (pow_bit0_pos_of_neg ha n)
-#align pow_bit1_neg pow_bit1_neg
```
Comment (eric-wieser) [inline Mathlib/Algebra/GroupPower/Order.lean:313]: Where did these lemmas go?

Comment (YaelDillies) [reply Mathlib/Algebra/GroupPower/Order.lean:313]: They were about deprecated declarations, so I simply removed them

**Comment by Vierkantor** [comment]
This looks good to me. @Ruben-VandeVelde, what exactly would you like to simplify in this PR?

**Comment by YaelDillies** [comment]
@Vierkantor, it was done. See #12288 and #12289

**Review by Ruben-VandeVelde — APPROVED** [review]
Apologies for the delay.

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**